app/scripts/controllers/transactions/index.js

```diff
@@ -558,20 +559,34 @@ export default class TransactionController extends EventEmitter {
     @returns {string} rawTx
   */
   async signTransaction(txId) {
+    const common = new Common({ chain: 'rinkeby', hardfork: 'berlin' });
+    console.log('signing');
     const txMeta = this.txStateManager.getTransaction(txId);
     // add network/chain id
     const chainId = this.getChainId();
-    const txParams = { ...txMeta.txParams, chainId };
+    const txParams = {
+      ...txMeta.txParams,
+      gasLimit: txMeta.txParams.gas,
+      chainId,
+    };
+    console.log('txParams', txParams);
     // sign tx
     const fromAddress = txParams.from;
-    const ethTx = new Transaction(txParams);
-    await this.signEthTx(ethTx, fromAddress);
+    const unsignedEthTx = TransactionFactory.fromTxData(txParams, {
+      common,
+      freeze: false,
```

Comment:
This shouldn't be strictly necessary, please test with and without to see if it works. By default transaction objects are frozen using Object.freeze making them immutable @rickycodes @darkwing